### PR TITLE
Chunk profiling stacktrace response

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we implement chunking for the get stacktraces API in the profiling plugin as typical response sizes are dozens of MB large.

### CODE REVIEW

**Comment by danielmitterdorfer** [comment]
@original-brownbear I've assigned you as a reviewer as you've implemented the underlying infrastructure and I'd appreciate if you could take a look. One remark about the test case: I did not inherit from `AbstractChunkedSerializingTestCase` because it would require to implement XContent parsing just for tests and our original base class was `AbstractWireSerializingTestCase` which is sufficient for our use case.

File: x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/GetProfilingResponse.java [hunk 1]
```diff
@@ -139,11 +140,6 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
-    @Override
-    public RestStatus status() {
-        return error != null ? ExceptionsHelper.status(ExceptionsHelper.unwrapCause(error)) : OK;
```
Comment (DaveCTurner) [inline x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/GetProfilingResponse.java:144]: Just checking that changing the status to always return `200 OK` is what you want to do. Today there's no way to return a different status alongside a chunked response, but we could add that feature if you would prefer. That said, most APIs will return `200 OK` even if there are errors within the response, or they'll just throw the exception.

**Review by DaveCTurner — COMMENTED** [review]
Looks good to me; I left one question.

**Review by original-brownbear — APPROVED** [review]
LGTM if you're ok with what David points out :)

File: x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/GetProfilingResponse.java (hunk 1 above)
Comment (DaveCTurner) [reply x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/GetProfilingResponse.java:144]: Discussed this elsewhere and agreed to stick to `200 OK` here.

**Review by DaveCTurner — COMMENTED** [review]
LGTM2

**Review by DaveCTurner — APPROVED** [review]

File: x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/GetProfilingResponse.java (hunk 1 above)
Comment (danielmitterdorfer) [reply x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/GetProfilingResponse.java:144]: As discussed offline the current behavior is fine for us. When an exception happens on the transport layer we would ultimately have `RestActionListener#onFailure()` send an appropriate response.

**Comment by danielmitterdorfer** [comment]
Thanks both for the quick review!

**Pull request merged**